### PR TITLE
feat: Ignore preserveUnknownFields in Diff from CRD Spec

### PR DESCRIPTION
**Pull request by subbuv26**
Signed-off-by: Subba Reddy Veeramreddy <subbareddyv.uoh@gmail.com>

Fixes https://github.com/argoproj/argo-cd/issues/6401
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: util/settings/settings.go [hunk 1]
```diff
@@ -550,12 +550,14 @@ func (mgr *SettingsManager) GetResourceOverrides() (map[string]v1alpha1.Resource
 	}
 
 	crdGK := "apiextensions.k8s.io/CustomResourceDefinition"
+	crdPrsvUnkn := "/spec/preserveUnknownFields"
```
Comment (subbuv26) [inline util/settings/settings.go:553]: Would it be good if these parameters declared as constants along with "status"

File: util/settings/settings.go [hunk 2]
```diff
@@ -550,12 +550,14 @@ func (mgr *SettingsManager) GetResourceOverrides() (map[string]v1alpha1.Resource
...
 
 	switch diffOptions.IgnoreResourceStatusField {
 	case "", "crd":
 		addStatusOverrideToGK(resourceOverrides, crdGK)
```
Comment (subbuv26) [inline util/settings/settings.go:557]: As the generic version of this function is available now, can this function be removed and use generic function instead.
Eg: 
`addIgnoreDiffItemOverrideToGK(resourceOverrides, "*/*", status)    // status = "/status"`

**Review by alexmt — APPROVED** [review]
LGTM.

I agree with your suggestions in the comments as well.

**Comment by yogeek** [comment]
Hello :) thanks for this improvement. Is it meant to be included in the 2.0.4 release or the next one please ?

**Comment by yogeek** [comment]
Follow-up to the previous question : 
I had a OutOfSync on a repo containing CRDs (argocd version 2.0.4 ) due to preserveUnknownFields
I saw that it was fixed in this PR merged on master.
So I updated my argocd deployment to point to latest docker image but I still have the OutOfSync  error ... did I miss something please ?
@alexmt

**Comment by pratiyushpushkar05** [comment]
@alexmt I updated argocd to v3.0.0 .I am seeing on CRD `redisenterpriseactiveactivedatabases.app.redislabs.com` of `redis-enterprises` going OutOfSync due to `preserveUnKnownFields: false` . Adding 
```
data:
  resource.customizations.ignoreDifferences.apiextensions.k8s.io_CustomResourceDefinition: |
    jsonPointers:
    - /spec/preserveUnknownFields
```
worked . Just wanted to know if this is the correct and only way .

**Comment by zhituha** [comment]
The same with 3.0.6

**Comment by zhituha** [comment]
That's why it's happened:
https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/2.14-3.0/#removing-default-ignores-of-preserveunknownfields-for-crd

**Pull request merged**